main/guides/zoe/contract-basics.md

```diff
@@ -1,46 +1,56 @@
-as part of the contract API by making it
-a method of the contract's `publicFacet`:
+The `greet` function, along with any other public function, must be made accessible through the `publicFacet` of the contract. The `publicFacet` is returned by the `start` function, which initializes the contract. In the greeting contract, the `start` function exposes the `greet` function by defining it as a method of the contract's `publicFacet`, as shown below:
 
 <<< @/../snippets/zoe/src/01-hello.js#publicFacet
 
-We mark it `Far(...)` to allow callers to use it from outside the contract
-and give it a suggestive interface name for debugging.
+We wrap the `greet` function in a `Far(...)` call to safely expose it as a remote object, accessible from outside the contract.
+We also give it a suggestive interface name `Hello` for debugging.
 _We'll discuss [Far in more detail](../js-programming/far) later._
 
 Putting it all together:
 
 <<< @/../snippets/zoe/src/01-hello.js#contract
 
+Let us save this code to a file named `01-hello.js` inside `src` directory. 
 ## Using, testing a contract
 
-Let's use some tests to explore how a contract is used.
+Before we deploy or start our contract, it is usually a good idea to run a few tests to validate its correctness. Agoric contracts are typically tested using the [ava](https://github.com/avajs/ava) framework. The test file begins with an `import @endo/init` to establish a [Hardened JavaScript](../js-programming/hardened-js) environment as below:
 
-Agoric contracts are typically tested
-using the [ava](https://github.com/avajs/ava) framework.
-They start with `@endo/init` to establish a [Hardened JavaScript](../js-programming/hardened-js) environment:
+<<< @/../snippets/zoe/contracts/test-zoe-01-hello.js#test-imports
 
-<<< @/../snippets/zoe/contracts/test-zoe-hello.js#test-imports
+We also import `E()` in order to make asynchronous method calls and `test` function from `ava`. _We'll talk more about [using `E()` for async method calls](../js-programming/eventual-send) later._ Also note the `eslint-disable-next-line` comment on line 3; this is needed in order to suppress an error that arises due to a known issue in `ava`.
 
-_We'll talk more about [using `E()` for async method calls](../js-programming/eventual-send) later._
+Following these `import` statements, we write a simple test that validates that the `greet` method works as expected as below:
 
-A test that the `greet` method works as expected looks like:
+<<< @/../snippets/zoe/contracts/test-zoe-01-hello.js#test1
 
-<<< @/../snippets/zoe/contracts/test-zoe-hello.js#test1
+
+Putting it all together:
+
+<<< @/../snippets/zoe/contracts/test-zoe-01-hello.js#test-01-hello
+
+Let's save this code in a file named `test-01-hello.js` in a `test` directory. Both `src` and `test` directories should lie in the same `contract` directory. Let us run the following command to execute the test:
+
+```sh
+npx ava --match="contract greets by name"
```

Comment:
The commands to run tests are discussed in the next section
https://docs.agoric.com/guides/zoe/contract-walkthru.html

Does it seem important to repeat them here?

Reply:
I do not see any significant overlap. Although, contract-walkthru needs a lot of fixing IMO but we will get to it once we have sorted out this and #1104.